tests/backup/backup_delete_test.go

```diff
@@ -0,0 +1,374 @@
+			}
+			log.Infof("List of backups - %v", backupNames)
+			log.Infof("List of Incremental backups Set 1 - %v", incrementalBackupNames)
+			log.Infof("List of Incremental backups Set 2 - %v", incrementalBackupNames2)
+		})
+		Step("Deleting incremental backup", func() {
+			log.InfoD("Deleting incremental backups")
+			backupDriver := Inst().Backup
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			for _, backupName := range incrementalBackupNames {
+				log.Infof("About to delete backup - %s", backupName)
+				backupUID, err := backupDriver.GetBackupUID(ctx, backupName, orgID)
+				log.FailOnError(err, "Failed while trying to get backup UID for - %s", backupName)
+				_, err = DeleteBackup(backupName, backupUID, orgID, ctx)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Deleting backup - [%s]", backupName))
+			}
+		})
+		Step("Check if backups are incremental backups or not", func() {
+			log.InfoD("Check if backups are incremental backups or not")
+			backupDriver := Inst().Backup
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx failed")
+			for _, incrementalBackupName := range incrementalBackupNames2 {
+				bkpUid, err := backupDriver.GetBackupUID(ctx, incrementalBackupName, orgID)
+				log.FailOnError(err, "Unable to fetch backup UID - %s", incrementalBackupName)
+				bkpInspectReq := &api.BackupInspectRequest{
+					Name:  incrementalBackupName,
+					OrgId: orgID,
+					Uid:   bkpUid,
+				}
+				bkpInspectResponse, err := backupDriver.InspectBackup(ctx, bkpInspectReq)
+				log.FailOnError(err, "Unable to fetch backup - %s", incrementalBackupName)
+				for _, vol := range bkpInspectResponse.GetBackup().GetVolumes() {
+					backupId := vol.GetBackupId()
+					dash.VerifyFatal(strings.Contains(backupId, "incr"), true,
+						fmt.Sprintf("Check if the backup %s is incremental or not", incrementalBackupName))
+				}
+			}
+		})
+		Step("Restoring the backed up namespaces", func() {
+			log.InfoD("Restoring the backed up namespaces")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			for index, namespace := range bkpNamespaces {
+				restoreName = fmt.Sprintf("%s-%s-%s", "test-restore", namespace, RandomString(4))
+				for strings.Contains(strings.Join(restoreNames, ","), restoreName) {
+					restoreName = fmt.Sprintf("%s-%s-%s", "test-restore", namespace, RandomString(4))
+				}
+				restoreNames = append(restoreNames, restoreName)
+				log.InfoD("Restoring [%s] namespace from the [%s] backup", namespace, backupNames[index])
+				err = CreateRestore(restoreName, backupNames[index], namespaceMapping, destinationClusterName, orgID, ctx, make(map[string]string))
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Creating restore [%s]", restoreName))
+			}
+		})
+		Step("Validate applications", func() {
+			log.InfoD("Validate applications")
+			ValidateApplications(contexts)
+		})
```

Comment:
Why do we need this? We would just be validating the apps deployed at the beginning of the test.

Reply:
Remove got accidentally  pasted twice.